Xcode 15.3 warning "capture of 'handle' with non-sendable type 'UnsafeMutablePointer<sourcekitd_request_handle_t?>'"

Running into this warning with Xcode 15.3 Beta 2 (15E5188j). Opening the issue to keep track of it. 

```
INFO: From Compiling Swift module @@sourcekitten~0.34.1//:SourceKittenFramework:
external/sourcekitten~0.34.1/Source/SourceKittenFramework/SourceKitObject.swift:125:39: warning: capture of 'handle' with non-sendable type 'UnsafeMutablePointer<sourcekitd_request_handle_t?>' (aka 'UnsafeMutablePointer<Optional<UnsafeMutableRawPointer>>') in a `@Sendable` closure
            sourcekitd_cancel_request(handle)
                                      ^
Swift.UnsafeMutablePointer:1:23: note: generic struct 'UnsafeMutablePointer' does not conform to the 'Sendable' protocol
@frozen public struct UnsafeMutablePointer<Pointee> : _Pointer {
```

--- a/Source/SourceKittenFramework/SourceKitObject.swift
+++ b/Source/SourceKittenFramework/SourceKitObject.swift
@@ -110,11 +110,11 @@ public final class SourceKitObject {
     }
 
     func sendAsync() async throws -> sourcekitd_response_t {
-        let handle = UnsafeMutablePointer<sourcekitd_request_handle_t?>.allocate(capacity: 1)
+        let handle = UncheckedSendable(UnsafeMutablePointer<sourcekitd_request_handle_t?>.allocate(capacity: 1))
 
         return try await withTaskCancellationHandler {
             try await withUnsafeThrowingContinuation { continuation in
-                sourcekitd_send_request(sourcekitdObject, handle) { response in
+                sourcekitd_send_request(sourcekitdObject, handle.value) { response in
                     enum SourceKitSendError: Error { case error, noResponse }
 
                     guard let response else {
@@ -130,7 +130,7 @@ public final class SourceKitObject {
                 }
             }
         } onCancel: {
-            sourcekitd_cancel_request(handle)
+            sourcekitd_cancel_request(handle.value)
         }
     }
 }

--- a/Source/SourceKittenFramework/UncheckedSendable.swift
+++ b/Source/SourceKittenFramework/UncheckedSendable.swift
@@ -1,0 +1,11 @@
+struct UncheckedSendable<Value>: @unchecked Sendable {
+    /// The unchecked value.
+    var value: Value
+
+    /// Initializes unchecked sendability around a value.
+    ///
+    /// - Parameter value: A value to make sendable in an unchecked way.
+    init(_ value: Value) {
+        self.value = value
+    }
+}

